suresh/loops.js

```diff
@@ -0,0 +1,317 @@
+//Loops for / while
+
+var arr = [1,2,3,4,5];
+
+//write a function that returns all odd numbers - [1,3,5]
+function oddNos (array) {
+  var oddNums = [];
+  var arrLength = array.length;
+
+  for (var i = 0; i < arrLength; i++) {
+ 	  if (array[i] % 2 !== 0) {
+  	  oddNums.push(array[i]);
+    }
+  }
+  return oddNums;
+}
+
+console.log("Odd numbers with For loop: ", oddNos(arr));
+
+function oddNosWithWhile (array) {
+  var oddNums = [];
+  var arrLength = array.length;
+  var i = 0;
+
+  while (i < arrLength) {
+ 	  if (array[i] % 2 !== 0) {
+  	  oddNums.push(array[i]);
+    }
+    i++;
+  }
+  return oddNums;
+}
+
+console.log("Odd numbers with While loop: ", oddNosWithWhile(arr));
+
+//write a function that returns all even numbers -[2,4]
+function evenNos (array) {
+  var evenNums = [];
+
+  for (var i = 0; i < arr.length; i++) {
+    if (array[i] % 2 === 0) {
+      evenNums.push(array[i]);
+    }
+  }
+  return evenNums;
+}
+
+console.log("Even numbers with For loop: ", evenNos(arr));
+
+function evenNosWithWhile (array) {
+  var evenNums = [];
+  var arrLength = array.length;
+  var i = 0;
+
+  while (i < arrLength) {
+ 	  if (array[i] % 2 === 0) {
+  	  evenNums.push(array[i]);
+    }
+    i++;
+  }
+  return evenNums;
+}
+
+console.log("Even numbers with While loop: ", evenNosWithWhile(arr));
+
+//write a function that returns the array backwards - [5,4,3,2,1];
+function reverse(array) {
+  var reverseArray = [];
+  var i = array.length-1;
+
+  for (i; i >= 0; i--) {
+    reverseArray.push(array[i]);
+  }
+  return reverseArray;
+}
+
+console.log("Reverse with For loop: ", reverse(arr));
+
+function reverseWithWhile (array) {
+  var reverseArray = [];
+  var i = array.length-1;
+
+  while (i >= 0) {
+ 	  reverseArray.push(array[i]);
+    i--;
+  }
+ return reverseArray;
+}
+
+console.log("Reverse with While loop: ", reverseWithWhile(arr));
+
+var arr1 = [6,7,8,9,10];
+//write a function that combines both arrays - [1,2,3,4,5,6,7,8,9,10]
+function combinesArray(arr1, arr2) {
+	var combArr = arr.slice();
```

Comment:
Can you think of another way to do this without using both slice and a for loop?

Reply:
added using concat